src/pages/index.js

```diff
@@ -67,55 +71,84 @@ imagePopup.setEventListeners();
 //User Information
 const nameSelector = '.profile__name';
 const descSelector = '.profile__desc';
-const profileElement = new UserInfo({name: nameSelector, desc: descSelector});
-
-//Edit modal
-const editForm = new PopupWithForm('.modal[id=editModal]',(inputs) => {
+const avatarSelector = '.profile__avatar-image';
+const profileElement = new UserInfo({name: nameSelector, desc: descSelector, avatar: avatarSelector});
+API.getUserInformation()
+    .then(userInfo => {
+        document.querySelector(avatarSelector).setAttribute('src', userInfo.avatar);
+        profileElement.setUserInfo({name: userInfo.name, desc: userInfo.about});
+    });
+
+//Edit user info modal
+const editInfoForm = new PopupWithForm('.modal[id=editModal]',(inputs) => {
     const userInfo = {
         name: inputs['name-input'],
-        desc: inputs['desc-input']
+        about: inputs['desc-input']
     }
 
-    profileElement.setUserInfo(userInfo);
+    return API.updateUserInformation({name: userInfo.name, about: userInfo.about})
+    .then((res) => {
+        profileElement.setUserInfo({name: res.name, desc: res.about});
+    });
 })
-editForm.setEventListeners();
+editInfoForm.setEventListeners();
 
+const editButtonSelector = '.profile__edit';
 const inputName = editModal.querySelector('.modal__input[name=name]');
 const inputDesc = editModal.querySelector('.modal__input[name=desc]');
-const editButton = document.querySelector('.profile__edit');
+const editButton = document.querySelector(editButtonSelector);
 editButton.addEventListener('click', openProfileModal);
 function openProfileModal() {
     const userInfo = profileElement.getUserInfo();
 
     inputName.value = userInfo.name;
     inputDesc.value = userInfo.desc;
-    editForm.open();
+    editInfoForm.open();
 }
 
 const editModalForm = document.forms['editModalForm'];
-const editModalValidator = new FormValidator(editModalForm, config);
+const editModalValidator = new FormValidator(editModalForm, config, editButtonSelector);
 editModalValidator.enableValidation();
 
+//Edit user avatar modal
+const editAvatarSelector = '.profile__avatar-container';
+const avatarImage = document.querySelector('.profile__avatar-image');
+const editAvatarForm = new PopupWithForm('.modal[id=editAvatarModal]', (inputs) => {
+    const linkInput = inputs['avatar-input'];
 
+    return API.updateAvatar(linkInput)
+    .then(res => {
+        avatarImage.setAttribute('src', res.avatar);
+    })
+})
+editAvatarForm.setEventListeners();
+
+const avatarElement = document.querySelector(editAvatarSelector);
+avatarElement.addEventListener('click', openEditAvatarModal)
+function openEditAvatarModal() {
+    editAvatarForm.open();    
+}
+
+const avatarModalForm = document.forms['editAvatarModalForm']
+const avatarModalValidator = new FormValidator(avatarModalForm, config, editAvatarSelector);
```

Comment:
**Could be improved**

You can **universally** create instances of validators for all forms in the project storing them inside one object `formValidators`.  And then you can take any validator using attribute `name` of the form where you need to disable/enable the submit button or remove errors.

```js
// define an object for storing validators
const formValidators = {}

const enableValidation = (config) => {
  const formList = Array.from(document.querySelectorAll(config.formSelector))
  formList.forEach((formElement) => {
    const validator = new FormValidator(config, formElement)
    // here you get the name of the form (if you don’t have it then you need to add it into each form in `index.html` first)
    const formName = formElement.getAttribute('name')

   // here you store the validator using the `name` of the form
    formValidators[formName] = validator;
    validator.enableValidation();
  });
};

enableValidation(config);

```

And now you can use them for disabling buttons or clearing errors:

```js

formValidators[ profileForm.getAttribute('name') ].resetValidation()

// or you can use a string – the name of the form (you know it from `index.html`)

formValidators['profile-form'].resetValidation()

```